java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Field.java

```diff
@@ -37,6 +40,7 @@
   private final String name;
   private final boolean nullable;
   private final ArrowType type;
+  private final Long dictionary;
```

Comment:
I'm not sure that the dictionary id has any use outside the stream/file formats. I was hesitant to mess with the stream/file encoding, so it seemed easier to add it here. I'm not sure of the repercussions either way...